node/overseer/src/lib.rs

```diff
@@ -1045,6 +1047,22 @@ struct SubsystemMeterReadouts {
 	signals: metered::Readout,
 }
 
+
+/// The amount of data to keep in the header freshness cache.
+/// With 32-byte block hashes and 6-second block times, this is enough
+/// to store known blocks extending back days.
+const LRU_CACHE_SIZE_BYTES: usize = 2 * 1024 * 1024;
+
+// This is a Zero-Sized ype which is used to communicate to the `known_leaves` LRU-cache the
+// size of the key.
+struct KnownLeafMarker;
```

Comment:
Why not use a regular LRU instead? The key if fixed-size so you can calculate the number of entries based on that. This impl seems hacky and we don't account for key sizes in other `MemoryLruCache` usages in runtime api caches.

Reply:
Also since we don't persist this data, this is not going to work with restarts, right?